lib/AST/ASTPrinter.cpp

```diff
@@ -755,7 +755,7 @@ class PrintAST : public ASTVisitor<PrintAST> {
   void printType(Type T) { printTypeWithOptions(T, Options); }
 
   void printTransformedTypeWithOptions(Type T, PrintOptions options) {
-    if (CurrentType && Current) {
+    if (CurrentType && Current && CurrentType->mayHaveMembers()) {
       if (T->hasArchetype()) {
         // Get the interface type, since TypeLocs still have
         // contextual types in them.
```

Comment:
I'm wondering if the mapTypeOutOfContext() is needed at all...

Reply:
It appears to be not. I’ll set up a follow-up PR removing it.